### DISCUSSION TITLE
load animation

### DISCUSSION THREAD

**Discussion by alex93skr** [k-ivan/visual-bookmarks-chrome#119] [Q&A]
@k-ivan Иван привет, подскажи пожалуйста, какой стиль отвечает за анимацию при открытии вкладки? Вот это плавное секундное появление фоновой картинки. Хочу его отключить через CSS, но не смог найти как.

**Comment by k-ivan** [comment]
@alex93skr Привет! Возможно имеет смысл сделать анимацию фона опцией(надо подумать будет над этим). 
В качестве быстрого решения, можно применить стили:
```css
.bg {
  transition: none;
  opacity: 1;
}
```
Важно: анимацию эти стили отключат, но если картинка загружается из урл адреса, ожидание подгрузки останется.

**Comment by alex93skr** [comment]
@k-ivan и еще небольшая просьба, тут же попрошу, что бы темы не плодить. Можешь добавить кнопку "найти" к форме поиска? По UI/IX она должна быть обязательно, да и в целом это очень нужно для юзабилити. Хоте бы просто ее туда добавь с { display: none }, но что бы она была физически, а дальше я уже сам дорисую стилями.

**Reply by k-ivan** [reply]
Хорошо, правда она будет относится только к поисковым системам(при поиске по закладкам событие срабатывает при наборе текста, поэтому смысла в ней нет). По срокам пока не скажу, как будет время.

**Comment by alex93skr** [comment]
@k-ivan спасибо за быстрый ответ, возможность менять стили это просто пушка. Огромное количество перебрал аналогов твоему расширению, но все решила именно эта фича. Для среднего пользователя, вряд ли она будет иметь решающее значение, но это они просто не понимают какие возможности оно открывает.

**Reply by k-ivan** [reply]
Очень рад, что нравится, но без фанатизма ), так как эта опция синхронизируется и ее объём ограничен браузером.

**Comment by k-ivan** [comment]
> @k-ivan и еще небольшая просьба, тут же попрошу, что бы темы не плодить. Можешь добавить кнопку "найти" к форме поиска? По UI/IX она должна быть обязательно, да и в целом это очень нужно для юзабилити. Хоте бы просто ее туда добавь с { display: none }, но что бы она была физически, а дальше я уже сам дорисую стилями.

Кнопка поиска появится в ближайшем обновлении.

**Comment by alex93skr** [comment]
@k-ivan Иван привет! Поздравляю с 5.6.0 - отличный релиз, все мелочи учтены. У меня она теперь выглядит вот так:

![image](https://user-images.githubusercontent.com/49789700/228208489-940464ff-c482-4c49-84f7-14ef365b85ac.png)

Вот на этих стилях:

`
.bg {
transition: none;
opacity: 1;
background-color: #a0a0a0;
}

header {
margin-top: 40px;
margin-bottom: 10px;
}

.bookmark__wrap:hover {
box-shadow: 0px 3px 10px rgba(0, 0, 0, 0.24);
}

.bookmark__action {
display: none;
}

.bookmark:nth-child(8),
.bookmark:nth-child(15),
.bookmark:nth-child(20),
.bookmark:nth-child(25) {
grid-column-start: 1;
}

.bookmark__title {
font-weight: 200;
}

.search-icon {
opacity: 0.5;
}

#searchSubmit {
width: 70px;
border-radius: 4px;
background-color: #ffcc00;
font-size: initial;
}

#searchSubmit:hover {
background-color: #f5c400;
}
`

Просто идеал. Никак только теперь не найду, где в магазине приложений хрома поставить шестую звезду)

**Comment by k-ivan** [comment]
@alex93skr Алекс привет! Спасибо! Выглядит отлично, мне нравится ☝️

**Comment by alex93skr** [comment]
@k-ivan Иван, ты уже придумал куда дальше развивать продукт? Потому что у меня есть еще пара идей. Например, прикрутить к поисковой строке функцию автодополнения запроса. У яндекса, насколько я понял, это можно делать через api предиктора https://yandex.ru/dev/predictor/

В инфинити https://chrome.google.com/webstore/detail/infinity-new-tab-pro/nnnkddnnlpamobajfibfdgfnbcnkgngh есть autocomplete predictions для всех поисковиков, можно дернуть алгоритм

**Reply by k-ivan** [reply]
Честно говоря не хотел в это расширение тащить этот функционал, он у меня есть в другом расширении, не хотелось бы повторяться.

**Reply by alex93skr** [reply]
@k-ivan Иван привет! По поводу autocomplete predictions для Яндекса, раз ты уже говоришь, что делал это в другом расширении, значит второй раз дело быстрее пойдет. Добавь хотя бы такой функционал в todo лист и как настроение будет, сделаешь его в visual-bookmarks. А мы пока будем ждать и надеяться 👍

**Reply by k-ivan** [reply]
Алекс привет! Дело не во времени, а в том, что у другого расширения должен быть какой-то отличительный функционал от этого, чтобы не было 2 расширений с одинаковым функционалом.

**Reply by k-ivan** [reply]
Я не буду утверждать точно, что это исключено. Бывали уже моменты, что говорил, что точно не буду что-то добавлять и потом передумывал. Но со стороны пользователя, я бы сильно этого не ждал.

**Reply by alex93skr** [reply]
Про какое "другое расширение" ты говоришь, у тебя есть еще один speed dial? Просто я вижу, что visual-bookmarks у тебя в репозитории самое звездное и думал, что бросаешь все силы на него.

**Reply by k-ivan** [reply]
Второе расширение, это не speed dial, это стартовая вкладка, и там тоже есть поле поиска и в нем есть автокомплит.  Оно вынесено в приватный репозиторий, поэтому в паблике его не видно.

**Reply by k-ivan** [reply]
В вебсторе можно перейти на его страничку, если кликнуть по табу похожие и на ссылку - другие расширения автора.

**Reply by alex93skr** [reply]
Посмотрел WanderTab и обзавидовался! Но честно говоря, не совсем понял сакраментальную идею расширения, как по мне очень сильный перегиб в сторону визуала при практической полном отсутствии функциональности. Вряд ли кто-то будет настраивать под себя вкладку, что бы открыть и просто смотреть на нее. По первой подумал, что это у тебя некая реинкарнация visual-bookmarks, но потом понял, что это самобытный продукт с совершенно другими целями. Хотя не ясно, зачем тратить столько сил на WanderTab для 90 человек, есть уже есть visual-bookmarks с готовой клиентской базой в 10к+, которые только и ждут, когда у них появится виджет погоды и автозаполнение в поиске 😎

**Reply by k-ivan** [reply]
Есть такое расширение, momentum, у которого тоже стартовая вкладка с прицелом в UI, и более **3 миллионов** пользователей. И большая часть кастомизации этого визуала спрятана за платной подпиской в отличии от wandertab. При этом wandertab уже сейчас превосходит momentum по части кастомизации интерфейса. Конечно, часть опций отсутствует и у моего расширения(пока, но это в основном всякие тудулисты). Я это к тому, что даже если кажется, что это всего лишь красивый стартовый экран, не факт, что это не будет людям интересно. Причем в случае с momentum люди даже платить готовы, чтобы получить дополнительную кастомизацию.

Ну и кол-во пользователей обусловлено тем, что расширение очень молодое и оно нигде и не кем не рекламировалось. Закладки родились аж в 2016 году. Количество сил выделяется обоим продуктам одинаково(даже закладкам больше, в силу большей сложности самого расширения).

**Comment by alex93skr** [comment]
@k-ivan Иван привет! Подскажи пожалуйста, вот эту огромную тень вверху каким стилем убрать? 

![image](https://user-images.githubusercontent.com/49789700/228833780-893e7f3c-c01b-4c94-b8d8-d401a14763ee.png)

**Reply by k-ivan** [reply]
Привет, один из способов:
```css
.has-image .bg::after  {
  display: none;
}
```
